Sources/Fullscreen/FullscreenGallery/FullscreenGalleryViewController.swift

```diff
@@ -0,0 +1,508 @@
+//
+// Copyright (c) 2019 FINN AS. All rights reserved.
+//
+
+import UIKit
+
+public protocol FullscreenGalleryViewControllerDelegate: class {
+    func fullscreenGalleryViewControllerDismissButtonTapped(_: FullscreenGalleryViewController)
+    func fullscreenGalleryViewController(_: FullscreenGalleryViewController, didSelectImageAtIndex: Int)
+}
+
+public protocol FullscreenGalleryViewControllerDataSource: class {
+    func fullscreenGalleryViewController(_: FullscreenGalleryViewController,
+                                         imageForUrlString urlString: String,
+                                         width: CGFloat,
+                                         completionHandler handler: @escaping (String, UIImage?, Error?) -> Void)
+}
+
+public class FullscreenGalleryViewController: UIPageViewController {
+
+    // MARK: - Public properties
+
+    public override var prefersStatusBarHidden: Bool {
+        return true
+    }
+
+    public weak var galleryDataSource: FullscreenGalleryViewControllerDataSource?
+    public weak var galleryDelegate: FullscreenGalleryViewControllerDelegate?
+
+    // MARK: - Private properties
+
+    private static let captionFadeDuration = 0.2
+    private static let dismissButtonSize: CGFloat = 30.0
+
+    private let viewModel: FullscreenGalleryViewModel
+    private let previewViewInitiallyVisible: Bool
+
+    private var currentImageIndex: Int
+    private var previewViewVisible: Bool
+    private var previewViewWasVisibleBeforePanning = false
+    private var hasPerformedInitialPreviewScroll = false
+
+    private var galleryTransitioningController: FullscreenGalleryTransitioningController? {
+        return transitioningDelegate as? FullscreenGalleryTransitioningController
+    }
+
+    // MARK: - UI properties
+
+    private lazy var backgroundView: UIView = {
+        let view = UIView(frame: .zero)
+        view.backgroundColor = .black
+        view.isUserInteractionEnabled = false
```

Comment:
Not sure if you need this one, it's `false` by default.

Reply:
It's actually `true` by default

Reply:
Yeah sorry, you're right. Don't mind me😅